plugins/newsdownloader.koplugin/epubdownloadbackend.lua

```diff
@@ -24,6 +24,50 @@ local max_redirects = 5; --prevent infinite redirects
 local TIMEOUT_CODE = "timeout" -- from socket.lua
 local MAXTIME_CODE = "maxtime reached" -- from sink_table_with_maxtime
 
+-- filter HTML using CSS selector
+local function filter(text, element)
+    local htmlparser = require("htmlparser")
+    local root = htmlparser.parse(text)
+    local filtered = nil
+    local selectors = {
+        "main",
+        "article",
+        "div#main",
+        "#main-article",
+        ".main-content",
+        "#body",
+        "#content",
+        ".content",
+        "div#article",
+        "div.article",
+        "div.post",
+        "div.post-outer",
+        ".l-root",
+        ".content-container",
+        ".StandardArticleBody_body",
+        "div#article-inner",
+        "div#newsstorytext",
+        "div.general",
+        }
+    if element then 
+        table.insert(selectors, 1, element)
```

Comment:
Just a comment to make sure that's your intended bahaviour: when the user has specified an element, you insert it at 1 (but you keep all the others at 2...19) - and you don't drop the table to use just the user-specified element.
So the other selectors are used as a fallback when the user element is not found. Right?

Reply:
Yeah, that is my point, I think it is a good safety net.

Reply:
I'm not quite sure to what extent I think a "safety net" makes sense. But you can always say enable_filter = false, correct?

Reply:
Yes, if someone does not want to filter then they are free to disable it. And it is the default anyway (as to not force changes upon people).